sycl/source/device_selector.cpp

```diff
@@ -77,6 +121,11 @@ device device_selector::select_device() const {
                 << "SYCL_PI_TRACE[all]: "
                 << "  device: " << DeviceName << std::endl;
     }
+    if (deviceType != info::device_type::all) {
+      std::cout
+          << "WARNING: the requested device and/or backend is not found.\n";
+      std::cout << PlatformName << " is chosen based on a heuristic.\n";
+    }
```

Comment:
Please, do not print any message by default. The standard way is to throw an exception. Please, print only if some special flag is enabled(e.g. detail::pi::trace(detail::pi::TraceLevel::PI_TRACE_BASIC))

Reply:
This message is printed ONLY when the user designate the specific device type as a parameter, but SYCL RT failed to find a device and chose a different device type based on a heuristic.
Actually, there was some complaint from the user about the heuristic based device selector.
Since now we allow the user to designate the desired device type, at least we should tell the user that the selected device is not what the user requested.
Silently choosing a different device type could be a problem for the user.
This message is to help the user to change the device type if they don't want the heuristic based approach.